internal/parser/translate_httproute.go

```diff
@@ -26,6 +26,15 @@ func fromHTTPRoutes(log logrus.FieldLogger, httpRouteList []*gatewayv1alpha2.HTT
 			hostnames = append(hostnames, kong.String(string(hostname)))
 		}
 
+		// validation for HTTPRoutes will happen at a higher layer, but in spite of that we run
+		// validation at this level as well as a fallback so that if routes are posted which
+		// are invalid somehow make it past validation (e.g. the webhook is not enabled) we can
+		// at least try to provide a helpful message about the situation in the manager logs.
+		if len(spec.Rules) < 1 {
```

Comment:
PS: it appears that such a check is added in 7aeae56ebe4052a7c8e84ae48cd392183b5857a3. Does it render this `if` redundant? If not, why not?

Reply:
The logic is currently to fail route generation altogether if _any_ invalid options are provided (this happens in `generateKongRoutesFromHTTPRouteRule()`). That function will produce an error if it can't produce at least one valid route so the condition you're bringing up is covered. I've added additional godoc comments to the function in df6143bd400d21d5f7b9178067220d0e3ed5d903 to account for this.